CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_DefaultActivities.uc

```diff
@@ -114,7 +117,104 @@ static function CreateInformantAssault (out array<X2DataTemplate> Templates)
 	
 	Templates.AddItem(ActivityAssault);
 }
+
+static function InformantAssaultSetup (XComGameState NewGameState, XComGameState_Activity ActivityState)
+{
+	local XComGameState_MissionSite MissionState;
+	local X2StrategyElementTemplateManager TemplateManager;
+	local X2RewardTemplate SoldierRewardTemplate;
+	local XComGameState_Reward RewardState;
+	local XComGameState_HeadquartersResistance ResHQ;
+
+	TemplateManager = class'X2StrategyElementTemplateManager'.static.GetStrategyElementTemplateManager();
+	SoldierRewardTemplate = X2RewardTemplate(TemplateManager.FindStrategyElementTemplate('Reward_Soldier'));
+	ResHQ = class'UIUtilities_Strategy'.static.GetResistanceHQ();
+
+	class'X2ActivityTemplate_Assault'.static.CreateMission(NewGameState, ActivityState);
+
+	foreach NewGameState.IterateByClassType(class'XComGameState_MissionSite', MissionState)
+	{
+		if (MissionState.ObjectID != ActivityState.PrimaryObjectRef.ObjectID)
+		{
+			MissionState = none;
+			continue;
+		}
+		
+		// Found the correct one
+		break;
+	}
+
+	if (MissionState == none)
+	{
+		`REDSCREEN("COULD NOT FETCH MISSION FROM CREATION STATE");
```

Comment:
```suggestion
		`REDSCREEN("CI InformantAssaultSetup: Could not fetch mission from creation state");
```

1) All caps doesn't help
2) Knowing **where** the error is from does help, especially as the message is very ambiguous on its own